cmd/kind/load/docker-image/docker-image.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package load implements the `load` command
+package load
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/kind/pkg/cluster"
+	clusternodes "sigs.k8s.io/kind/pkg/cluster/nodes"
+	"sigs.k8s.io/kind/pkg/docker"
+)
+
+type flagpole struct {
+	Name  string
+	Nodes []string
+}
+
+// NewCommand returns a new cobra.Command for loading an image into a cluster
+func NewCommand() *cobra.Command {
+	flags := &flagpole{}
+	cmd := &cobra.Command{
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return errors.New("name of image is required")
+			}
+			return nil
+		},
+		Use:   "docker-image",
+		Short: "loads docker image from host into nodes",
+		Long:  "loads docker image from host into all or specified nodes by name",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runE(flags, cmd, args)
+		},
+	}
+	cmd.Flags().StringVar(
+		&flags.Name,
+		"name",
+		cluster.DefaultName,
+		"the cluster context name",
+	)
+	cmd.Flags().StringSliceVar(
+		&flags.Nodes,
+		"nodes",
+		nil,
+		"comma seperated list of nodes to load images into",
+	)
+	return cmd
+}
+
+func runE(flags *flagpole, cmd *cobra.Command, args []string) error {
+	imageTarName := "image.tar"
```

Comment:
possibly this should be a tempfile that we then defer removing?

Reply:
Is there a good spot to work out of? I tried writing to `/tmp` but the `docker cp` wouldn't work. It wouldn't throw an error, just wouldn't copy.

Reply:
Was this on Mac? The tempdir wrapper in pkg/fs works around this, we may need a tempfiles wrapper too

Reply:
https://github.com/kubernetes-sigs/kind/blob/e14d2d27103b4d1b4a2e4ba81d08a672cf64dec1/pkg/fs/fs.go#L30-L43

Reply:
Just to clarify which I think I understand now is you are looking for a temp file on the `host`. I was trying to copy into a tmp folder inside the container. 

I agree the host should be done, should we also have a temp place inside the container to copy to? (e.g. `docker cp /tmp/img.tar kind-1-worker3:/tmp/img.tar`)

When I tried to copy to `/tmp` in the container, that's where I was getting the silent failure.

Reply:
🤦‍♂️

Reply:
This comment should be on the imageTarName below, not this destinationImageTar

Reply:
Within the container this is fine for now. We should similarly clean it up after the load though. (Or see about piping it to the load and never writing the tar to disk directly within the container)